<a href="https://colab.research.google.com/github/Chaitanyachafle/Portfolio/blob/main/python_case_study_1_nps_calculation.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Que-1 Calculate the NPS for company.
Rating below 6 called as defaulter
Rating 7 & 8 called as Passive
Rating above 6 called as Promoter
Formula : NPS = % Promoter - % Defaulter

In [ ]:
import numpy as np

In [ ]:
!gdown 1c0ClC8SrPwJq5rrkyMKyPn80nyHcFikK

Downloading...
From: https://drive.google.com/uc?id=1c0ClC8SrPwJq5rrkyMKyPn80nyHcFikK
To: /content/survey.txt
100% 2.55k/2.55k [00:00<00:00, 12.2MB/s]


In [ ]:
score = np.loadtxt("survey.txt",dtype = "int")

In [ ]:
score.ndim # Checking Dimension in array

1

In [ ]:
score.shape

(1167,)

In [ ]:
total_count= len(score) # Giving us total count to rating in sample set
print(total_count)

1167


In [ ]:
# calculating defaulter and Promoter count respectively

In [ ]:
num_defaulter=len(score[score <6 ])/total_count
print(num_defaulter)

0.28449014567266495


In [ ]:
num_promoter=len(score[score >=9 ])/total_count
print(num_promoter)

0.5218508997429306


In [ ]:
#Calculating NPS

In [ ]:
NPS = (num_promoter - num_defaulter) * 100
print("NPS-",NPS)

NPS- 23.73607540702657


Remark:A Net Promoter Score (NPS) of 23.74 is a positive score, but it's essential to understand what this score means in the context of your industry and customer expectations
1. Follow Up with Detractors
2. Enhance Customer Support
3. Product service Imporvemnet
4. Employee engagement activity